lore/models/sklearn.py

```diff
@@ -1,4 +1,45 @@
+from sklearn.model_selection import RandomizedSearchCV
+
 import lore.models.base
 
+
 class Base(lore.models.base.Base):
-    pass
+    
+    def hyper_parameter_search(
+        self,
+        param_distributions,
+        n_iter=10,
+        scoring=None,
+        fit_params={},
+        n_jobs=1,
+        iid=True,
+        refit=True,
+        cv=None,
+        verbose=0,
+        pre_dispatch='2*njobs',
+        random_state=None,
+        error_score='raise',
+        return_train_score=True
+    ):
+        result = RandomizedSearchCV(
+            self.estimator,
+            param_distributions,
+            n_iter=n_iter,
+            scoring=scoring,
+            n_jobs=n_jobs,
+            iid=iid,
+            refit=refit,
+            cv=cv,
+            verbose=verbose,
+            pre_dispatch=pre_dispatch,
+            random_state=random_state,
+            error_score=error_score,
+            return_train_score=return_train_score
+        ).fit(
+            self.pipeline.encoded_training_data.x,
+            y=self.pipeline.encoded_training_data.y,
+            **fit_params
+        )
+        self.estimator.sklearn = result.best_estimator_
```

Comment:
Odd that you pass `self.estimator` in params, but assign `self.estimator.sklearn` from the result. This seems like it might be double wrapping, since `lore.estimators.SKLearn` inherits from `sklearn.base.BaseEstimator`. Am I missing something?

Reply:
https://github.com/instacart/lore/blob/master/lore/estimators/sklearn.py#L14 `fit` and `predict` are called on the `sklearn` property.
`estimator` here is the lore estimator class, not the actual SKLearn estimator. This wrapping https://github.com/instacart/lore/blob/master/lore/estimators/sklearn.py is indeed almost useless if it wasn't for the way arguments are passed here https://github.com/instacart/lore/blob/master/lore/models/base.py#L48

Reply:
I think this might leave self.estimator without the correct hyperparams from best_estimator_, since you're assigning it to the member. Would be good to see a test case.

Reply:
We could override the `fit` method in here https://github.com/instacart/lore/blob/master/lore/models/sklearn.py and the whole SKLearn estimator lore class would be useless.

Reply:
You've got a point, the meagre value it adds is the estimator adds is timed logging calls to fit and predict.

Reply:
My preference would just be to unname the `x` param in the model base class call to fit, since it's the first param for all. Then we can delete the sklearn estimator wrapper class.